waku/waku_sync/raw_bindings.nim

```diff
@@ -5,174 +5,129 @@ else:
-
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L42
-proc constructNegentropy(storage: Storage, frameSizeLimit: culong): Negentropy {.importcpp: "negentropy::Negentropy(@)", constructor.}
+proc constructNegentropy(storage: pointer, frameSizeLimit: uint64): pointer {.header: NEGENTROPY_HEADER, importc: "negentropy_new".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L46
-proc raw_initiate(this: Negentropy): String {.importcpp: "negentropy::initiate".}
+proc raw_initiate(negentropy: pointer): cstring {.header: NEGENTROPY_HEADER, importc: "negentropy_initiate".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L58
-proc raw_setInitiator(this: Negentropy) {.importcpp: "negentropy::setInitiator".}
+proc raw_setInitiator(negentropy: pointer) {.header: NEGENTROPY_HEADER, importc: "negentropy_setinitiator".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L62
-proc raw_reconcile(this: Negentropy, query: String): String {.importcpp: "negentropy::reconcile".}
+proc raw_reconcile(negentropy: pointer, query: ptr Buffer): cstring {.header: NEGENTROPY_HEADER, importc: "reconcile".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L69
-proc raw_reconcile(this: Negentropy, query: String, haveIds: var Vector[String], needIds: var Vector[String]): String {.importcpp: "negentropy::reconcile".}
+proc raw_reconcile(negentropy: pointer, query: ptr Buffer, haveIds: cstringArray, haveIdsCount: pointer, needIds: cstringArray, needIdsCount: pointer): cstring {.header: NEGENTROPY_HEADER, importc: "reconcile_with_ids".}
 
 ### Wrappings ###
 
-proc new*(T: type Storage): T =
-  let storage = Storage()
+#TODO: Change all these methods to private as we don't want them to be exposed outside Sync package
+proc new_storage*(): pointer =
+  let storage = storage_init("", "")
 
   return storage
 
-proc erase*(self: Storage, id: int64, hash: WakuMessageHash) =
-  let cppString = String.fromWakuMessageHash(hash)
+proc erase*(storage: pointer, id: int64, hash: WakuMessageHash): bool =
+  let cString = toBuffer(hash)
   
-  self.raw_erase(clong(id), cppString)
+  return raw_erase(storage, uint64(id), cString.unsafeAddr)
 
-proc insert*(self: Storage, id: int64, hash: WakuMessageHash) =
-  let cppString = String.fromWakuMessageHash(hash)
+proc insert*(storage: pointer, id: int64, hash: WakuMessageHash): bool =
+  let cString = toBuffer(hash)
   
-  self.raw_insert(clong(id), cppString)
+  return raw_insert(storage, uint64(id), cString.unsafeAddr)
 
-proc new*(T: type Negentropy, storage: Storage, frameSizeLimit: uint64): T =
-  let negentropy = constructNegentropy(storage, culong(frameSizeLimit))
+proc new_negentropy*(storage: pointer, frameSizeLimit: uint64): pointer =
```

Comment:
I thinking we'll need objects wrapping storage and negentropy. `pointer` are not super ergonomic.